File: Projects/Dagger/Projects/Project-3-ConstructorInjection/app/src/main/java/com/droid/code/MyApplication.java
package com.droid.code;

import android.app.Application;
import android.util.Log;

import com.droid.code.data.local.DatabaseService;
import com.droid.code.data.remote.NetworkService;
import com.droid.code.di.components.ApplicationComponent;
import com.droid.code.di.components.DaggerApplicationComponent;
import com.droid.code.di.modules.ApplicationModule;

import javax.inject.Inject;


public class MyApplication extends Application {

    @Inject
    public NetworkService networkService;

    @Inject
    public DatabaseService databaseService;


    public ApplicationComponent applicationComponent;

    @Override
    public void onCreate() {
        super.onCreate();

        // ---> This is used to store a application level instance
        // And later we can use it in activity to create activity level instances

        // ---> This is used for creation of application level instances
        applicationComponent =  DaggerApplicationComponent.builder()
                // Module object creation
                .applicationModule(new ApplicationModule(this))
                // Building the modules
                .build();

        applicationComponent.inject(this);

        Log.d("Tag","Dependencies Added");
    }
}

File: Projects/Dagger/Projects/Project-5-DaggerInKotlin/app/src/main/java/com/demo/kotlin/sample/MyApplication.kt
package com.demo.kotlin.sample

import android.app.Application

import com.demo.kotlin.sample.data.local.DatabaseService
import com.demo.kotlin.sample.data.remote.NetworkService
import com.demo.kotlin.sample.di.component.ApplicationComponent
import com.demo.kotlin.sample.di.component.DaggerApplicationComponent
import com.demo.kotlin.sample.di.module.ApplicationModule

import javax.inject.Inject

class MyApplication : Application() {

    lateinit var applicationComponent: ApplicationComponent

    @Inject
    lateinit var networkService: NetworkService

    @Inject
    lateinit var databaseService: DatabaseService

    override fun onCreate() {
        super.onCreate()
        getDependencies()
    }

    private fun getDependencies() {
        applicationComponent = DaggerApplicationComponent
                .builder()
                .applicationModule(ApplicationModule(this))
                .build()
        applicationComponent.inject(this)
    }
}